src/error-handling.ts

```diff
@@ -1,58 +1,160 @@
+			detail: improvedErrors.message,
+			modal: true,
+		});
+	} else {
+		showErrorDialog((actionTitle ? `FAILED ${actionTitle.toLocaleUpperCase()}: ` : '') + improvedErrors.message);
+	}
 
-	// Log Axios request URL
+	// Log to console
 	if (err instanceof AxiosError) {
-		e += ' Requested ' + err.request?.method + ' ' + err.request?.path;
+		// Log Request CRUD + path
+		log(
+			'error',
+			'Rejected the Make request ' +
+				(err.request?.method || '').toUpperCase() +
+				' ' +
+				err.request?.path +
+				(err.response?.data instanceof Object ? '\n\t' + JSON.stringify(err.response?.data) : '') +
+				(typeof err.response?.data === 'string' ? '\n\t' + err.response?.data : ''),
+		);
+	} else if (err instanceof Error && !improvedErrors.isImproved) {
+		// Log to VS Code console
+		log('error', improvedErrors.message);
+		if (!improvedErrors.isImproved) {
+			log('error', `\tSTACK: ${err.stack}`);
+		}
 	}
+}
 
-	// Show error message in VS Code
-	vscode.window.showErrorMessage('ERROR: ' + e);
+/**
+ * Alias for old usages in JS files.
+ * @deprecated Use `showAndLogError()` name instead.
+ */
+export const showError = showAndLogError;
 
-	// Log to VS Code console
-	log('error', e, false);
+/**
+ * Shows the error message in VS Code toast notification.
+ *
+ * Note: Does not log the error into error console. If needed to log it, use `showAndLogError()` instead.
+ */
+export function showErrorDialog(message: string, messageOptions: vscode.MessageOptions = {}): void {
+	vscode.window
+		.showErrorMessage(
+			message,
+			messageOptions,
+			{ title: 'OK' },
+			{ title: 'Show error log', isCloseAffordance: true },
+		)
+		.then((userResponse) => {
+			if (userResponse?.title === 'Show error log') {
+				showLog();
+			}
+		});
 }
 
-
 /**
  * Function wrapper, which catches errors in async functions and displays them in the VSCode window.
  */
-export function catchError<T extends (...args: any[]) => Promise<any>>(
-	errorTitle: string,
-	asyncFunc: T
-): T {
-	return <T>(async (...args:any[]) => {
+export function catchError<T extends (...args: any[]) => Promise<any>>(errorTitle: string, asyncFunc: T): T {
+	return <T>(async (...args: any[]) => {
 		try {
 			return await asyncFunc(...args);
 		} catch (err: any) {
-			showError(err, errorTitle);
+			showAndLogError(err, errorTitle);
 		}
 	});
-  }
+}
+
+/**
+ * Improves the specific error message to more user friendly one. In all other cases returns undefined.
+ */
+function getImprovedErrorMessage(message: string): string | undefined {
+	if (message === "Value doesn't match pattern in parameter 'name'.") {
+		return 'Some local component has the invalid ID (name) defined. The ID must to follow specific restrictions for used letters, number and symbols.';
+	}
+	if (message.includes(' in JSONC at position ')) {
+		return 'JSON has corrupted structure. Code cannot be deployed. Find and fix the error and try to deploy again.';
```

Comment:
nevim zda uzivatel chape, ze dela "deploy" ... myslim, ze spise dela "save changes"

Reply:
Zatim je to nejlepsi pojmenovani, co se vymyslelo... i kdyz chapu, ze neni idealni

Deploy = upload k nam do cloudu
Save = Ulozeni lokalniho souboru lokalne.

... Ale muzeme to kdykoliv prejmenovat lip, kdyz najdeme lepsi variantu